packages/jira-adapter/src/filters/workflow/workflow_modification_filter.ts

```diff
@@ -181,58 +181,84 @@ const deployWorkflowModification = async ({
   await cleanTempInstance()
 }
 
-const filter: FilterCreator = ({ client, config, elementsSource, paginator }) => ({
-  name: 'workflowModificationFilter',
-  onFetch: async elements => {
-    if (!config.client.usePrivateAPI) {
-      log.debug('Skipping workflow modification filter because private API is not enabled')
-
-      return
-    }
-    const workflowType = findObject(elements, WORKFLOW_TYPE_NAME)
-    if (workflowType !== undefined) {
-      workflowType.annotations[CORE_ANNOTATIONS.UPDATABLE] = true
-      await addAnnotationRecursively(workflowType, CORE_ANNOTATIONS.UPDATABLE)
+const getWorkflowSchemeIds = (elementsSource: ReadOnlyElementsSource): () => Promise<ElemID[]> => {
+  let ids: ElemID[]
+  let workflowSchemesPromise: Promise<ElemID[]>
+  return async (): Promise<ElemID[]> => {
+    if (ids === undefined) {
+      if (workflowSchemesPromise === undefined) {
+        workflowSchemesPromise = awu(await elementsSource.list())
+          .filter(id => id.typeName === WORKFLOW_SCHEME_TYPE_NAME)
+          .filter(id => id.idType === 'instance')
+          .toArray()
+      }
+      ids = await workflowSchemesPromise
```

Comment:
I don't think the ids is needed here. I think we can do
```suggestion
const getWorkflowSchemeIds = (elementsSource: ReadOnlyElementsSource): () => Promise<ElemID[]> => {
  let workflowSchemesPromise: Promise<ElemID[]>
  return (): Promise<ElemID[]> => {
      if (workflowSchemesPromise === undefined) {
        workflowSchemesPromise = awu(await elementsSource.list())
          .filter(id => id.typeName === WORKFLOW_SCHEME_TYPE_NAME)
          .filter(id => id.idType === 'instance')
          .toArray()
      }
      return workflowSchemesPromise
```